engines/dfc_provider/app/services/quantitative_value_builder.rb

```diff
@@ -69,34 +69,38 @@ def self.apply(quantity, product)
   def self.map_unit(unit) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength
     quantity_unit = DfcLoader.connector.MEASURES.UNIT.QUANTITYUNIT
 
+    # The unit name is only set for items. The name is implied for weight and
+    # volume and filled in by `WeightsAndMeasures`.
```

Comment:
👋 
Something like that, yes. I think the idea is that it is able to understand the relation between units, so if I buy 10x 100g, it can display "1kg".

Reply:
I'm not sure about that example. I think that a line item would still say 10 x 100g. It's more clear. But if you have a product measured in weight and a unit scale of 1000 then it will automatically display as kg. But if you have other variants with smaller values then you may want to choose a unit scale of 1 and then it's displayed as g.

That's the common case. As admin, you choose _kg_ from a select and it sets the scale to 1000. But there's also a case where the best scale is chosen as in David's example but I didn't follow the code path. I'm not sure in which part of the app that is applied.

Reply:
Thanks for the clarification !